### PR TITLE
[SYCL] [L0] Fix for unreferenced parameters.

### PR DESCRIPTION
**Pull request by rdeodhar**
Adds references to parameters to avoid compilation errors.

### CODE REVIEW

File: sycl/source/queue.cpp [hunk 1]
```diff
@@ -25,6 +25,7 @@ __SYCL_INLINE_VER_NAMESPACE(_V1) {
 queue::queue(const context &SyclContext, const device_selector &DeviceSelector,
              const async_handler &AsyncHandler, const property_list &PropList,
              Discriminator Disc) {
```
Comment (bmyates) [inline sycl/source/queue.cpp:27]: could do `[[maybe_unused]] Discriminator Disc`

Comment (bader) [reply sycl/source/queue.cpp:27]: I don't see this parameter in the spec - https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_queue_interface. So we added a parameter, which we ignore. Could you please clarify why do we ignore `Disc`?

**Comment by bader** [comment]
> [SYCL] [L0] Fix for unreferenced parameters.

I see that this patch changes SYCL queue constructor in the common part of the runtime library, so it's not clear why this change is tagged with `[L0]`. It's not specific to Level Zero, right?

File: sycl/source/queue.cpp (hunk 1 above)
Comment (rdeodhar) [reply sycl/source/queue.cpp:27]: See comment in #7196.

**Review by againull — APPROVED** [review]

**Pull request merged**